RO-2712: Fjern setting overrides, sett mapCenter til default av på web + fiks liten bug

jorgkv opened the pull request:
- Setter kartsenter til default av på web
- Fjerner override-kode
  - Måtte gjøre om litt på oppstartskoden i userSettingService pga dette
- Fikser en liten feil i userSettingService som gjorde det mulig å overskrive lagrede settings. Vet ikke om det faktisk skjedde, men det var teoretisk mulig, lagde en enhetstest for å bevise det.

jorgkv committed:
Fiks user setting overrides og sett mapCenter til default av på web

gruble commented on src/app/core/services/user-setting/user-setting-overrides.ts:
```diff
@@ -0,0 +1,44 @@
+import { Capacitor } from '@capacitor/core';
```
Kan du legge inn en kort beskrivelse av hva som er vitsen med denne fila øverst?

gruble commented on src/app/core/models/user-settings.model.ts:
```diff
@@ -25,7 +25,7 @@ export interface UserSetting extends InfoPopupSettings {
   copyright?: string;
   photographer?: string;
-  lastOverridden?: Date;
+  lastOverridden?: number;
```
Bruk gjerne JsDoc for å dokumentere denne også, slik vi har gjort for de nye innstillingene under

gruble commented on src/app/core/services/user-setting/user-setting.service.ts:
```diff
@@ -414,17 +414,7 @@ export class UserSettingService extends NgDestoryBase implements OnReset {
-        };
-        return updatedSettings as UserSetting;
-      }),
+      map((result) => applyUserSettingOverrides(result, USER_SETTINGS_OVERRIDES)),
```
Kommentaren over denne linja bør oppdateres

gruble reviewed (commented):
Fint at kartsenter ikke er på for web.

Men jeg forstår ikke hvorfor vi trenger å holde styr på når standard-innstillingene endres?
Hvis du mener dette er noe vi trenger, ta den gjerne inn, men kunne ønsket meg bedre dokumentasjon på dette.

gruble reviewed (changes requested):
Fint at kartsenter ikke er på for web.

Jeg forstår ikke hvorfor vi trenger å holde styr på når standard-innstillingene endres?
Hvis du mener dette er noe vi trenger, ta den gjerne inn. 
Kanskje noe bedre dokumentasjon på dette vil gjøre at selv jeg skjønner det?

jorgkv commented:
Takk for review! Du har helt rett, tror ikke vi trenger holde styr på det. Kommentaren din gjorde at jeg oppdaget en bug også, tror jeg. Skal fikse ved ledig tid 👍

jorgkv committed:
Legg til dokumentasjon og rydd opp

jorgkv committed:
Slett override-kode, fiks en annen liten feil

  - Oppdaget at update-metoden i servicen som ble lagt til for en tid tilbake kunne risikere å føre til overskriving av data
  - Har lagt til en enhetstest for dette og fiksa feilen ved at lagring baserer seg på det fulle userSettings$-objektet og ikke bare userSettingsInMemory

jorgkv committed:
Fjern setting ikke lenger i bruk

gruble reviewed (approved):
Mye enklere å forstå og funker utmerket på web i hvert fall. Har ikke testet i app.
